tf_encrypted/keras/models/sequential.py

```diff
@@ -112,15 +113,21 @@ def set_weights(self, weights, sess=None):
           matching the output of model.get_weights()
       sess: tfe session"""
 
+    if not sess:
+      sess = tfe.Session()
+      # Set as keras global session so the
+      # model can be run with K.get_session():
+      K.set_session(sess)
+      # Initialize model weights before setting weights
+      sess.run(tf.global_variables_initializer())
```

Comment:
When/if we move to a `set_session` approach then all `sess.run(tf.global_variables_initializer())` are dangerous, as they will potentially override existing values.

Two things to keep in mind:

1. Variables that have been explicitly assigned to do not need to initialised AFAICT
2. It might be better to keep track of each variable used in TFE Keras (this is done in TF Keras as well) and whether or not it has been initialised; setting the variable should also mark it as initialised, per 1.

In this approach variables are initialised when the model is first used.

Reply:
ah you are right. I ran some quick tests. If a variable has been explicitly assigned we don't need to initialize it.

Reply:
I made couple updates:
- added get_session and set_session. For get session maybe there is a better approach other than checking if K.get_session() returns a tfe.Session
- When passing to set_weights a list of private variables, I checked if they have been initialized with `is_variable_initialized`. If not, we initialize them.  I am doing that so in [`test_weights_as_private_var`](https://github.com/tf-encrypted/tf-encrypted/blob/756547ac32473aea982b51e6c1899ee534e1725a/tf_encrypted/keras/models/sequential_test.py#L99) we don't have to manually create a tfe.Session to initialize the new weights then pass around the tfe.Session to `set_weights` etc.
- Note: model variables get tracked in [layer weights attribute](https://github.com/tf-encrypted/tf-encrypted/blob/756547ac32473aea982b51e6c1899ee534e1725a/tf_encrypted/keras/engine/base_layer.py#L111). I am not tracking if they have been initialized or not because variables that have been explicitly assigned to do not need to be initialized. If we need to check if the variable has been initialized, we could use is_variable_initialized`.